pkg/controllers/management/auth/user.go

```diff
@@ -204,6 +210,33 @@ func (l *userLifecycle) Remove(user *v3.User) (runtime.Object, error) {
 		return nil, err
 	}
 
+	// if tokens exist, ClusterUserAttrib may have been created
+	if len(tokens) > 0 {
+		clusters, err := l.clusterLister.List("", labels.Everything())
+		if err != nil {
+			return nil, err
+		}
+		// find the intersection of cluster.Names with token.ClusterName
+		set := make([]*v3.Cluster, 0)
+		for _, cluster := range clusters {
```

Comment:
If the token already has a clusterName on it, why don't we just get the clusters explicitly?